python/foglamp/services/core/api/configuration.py

```diff
@@ -22,12 +25,16 @@
     | GET            | /foglamp/category/{category_name}                           |
     | GET POST PUT   | /foglamp/category/{category_name}/{config_item}             |
     | DELETE         | /foglamp/category/{category_name}/{config_item}/value       |
+    | POST           | /foglamp/category/{category_name}/{config_item}/upload      |
```

Comment:
@MarkRiddoch I have added an additional endpoint to upload script type and works only if it has category name and config item exists in database. Is it okay? Or should we need to handle in `PUT   | /foglamp/category/{category_name}/{config_item}` itself.

And for add service and add task we have an optional `config` data parameter during this if any of config item has type `script` then it should not work at the moment.

Also for fresh category and merge category if any of config item has type `script` then it also not work at this moment.
Do we really need to handle these situation?
Please advise.

Reply:
I need to think about how best to handle the merge and add_service/task should work with this. My suggestion is let's get what we have in place as this let's us move forward and solve these other issues, if we need to, in a future Jira ticket.

Reply:
@MarkRiddoch See https://github.com/foglamp/FogLAMP/pull/1186 if it looks good to you.